doc/news/changes/incompatibilities/20200521Maier-2

```diff
@@ -0,0 +1,9 @@
+Removed: The <code>DEAL_II_WITH_CXX14</code> and
+<code>DEAL_II_WITH_CXX17</code> configuration options have been removed.
+The library will now be compiled with the defaultr C++ standard enabled by
+the compiler. This is (as of May 2020) C++14 for all compilers. If you want
+to override that behavior, please set the C++ standard directly for example
+by configuring with <code>-DDEAL_II_CXX_FLAGS="-std=c++17"</code>, or by
+setting the environement variable <code>CXXFLAGS="-std=c++17".
```

Comment:
Should we tell people to simply use [-DCMAKE_CXX_STANDARD](https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html#variable:CMAKE_CXX_STANDARD) instead of using our own custom flag. I think that there is value in using regular CMake flag when possible.

Reply:
That would be a good option - in any case, this PR is a good progress towards that goal. Or do I misunderstand you that you do not like having the user giving `-std=c++17`? While we could encourage the use of the CMAKE mechanism, it would not be our own custom flag any more, but the user's flag.

Reply:
I am totally fine with the current PR.

Reply:
The problem with `CMAKE_CXX_STANDARD` is that it only sets the C++ standard used for targets. I.e., no check we run is affected by this. Also, `CMAKE_CXX_STANDARD` actually does not guarantee that a certain standard is actually set... fun.

Reply:
?? I use 
```
SET (CMAKE_CXX_STANDARD 11)
SET (CMAKE_CXX_STANDARD_REQUIRED ON)
SET (CMAKE_CXX_EXTENSIONS OFF)
```
in one of my projects and that puts `-std=c++11` on the compiler command line.

Reply:
@bangerth Yes, it does but only when compiling a target because it sets a "taget property".

Reply:
So why not set it as a property on all targets then?

Reply:
The problem is that I don't see any sane way of running all our language feature and regression tests for that case (`CMAKE_CXX_STANDARD` does not influence code compiled via `CHECK_CXX_SOURCE_COMPILES`).

Reply:
I see. That's a shame, but I know you know this stuff better than anyone else...